dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/datastore/DatastoreFileWriter.kt

```diff
@@ -71,6 +72,17 @@ internal class DatastoreFileWriter(
         }
     }
 
+    @WorkerThread
+    internal fun clearAllData() {
+        val dataStoreDirectory = dataStoreFileHelper.createDataStoreDirectoryIfNecessary(
```

Comment:
this is a bit confusing: why do we need to create a directory (if it is not present) just to make it empty? Also `deleteDirectoryContentsSafe` will make `number_of_files` I/O calls, while we can just do 1 I/O call - delete the directory itself.

Maybe it is better just to get a `File` object pointing to the directory (just an object, without creating directory), check if it exists and then just call `delete` if it is the case?

Reply:
Are you sure that it's possible in one call? The documentation for `delete` indicates that the directory [must be empty](https://developer.android.com/reference/java/io/File#delete()) to be deleted

Reply:
if it doesn't work I would just create a method there that checks if a dir exists by path and it deletes all the files inside recursively. As @0xnm mentions we should not create a dir just to be able to call that method.

Reply:
I was just reusing an existing method that does both. I'll split it, so that one method just returns the path and another also creates if necessary and then we won't have an issue with this.